mlf_core/create/templates/mlflow/mlflow_pytorch/{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug_no_hyphen }}/{{ cookiecutter.project_slug_no_hyphen }}.py

```diff
@@ -1,95 +1,86 @@
-    set_general_random_seeds(general_seed)
-    set_pytorch_random_seeds(pytorch_seed, use_cuda=use_cuda)
-
-    # Load training and testing data
-    train_loader, test_loader = load_train_test_data(training_batch_size, test_batch_size)
-
-    # Define model, device and optimizer
-    if torch.cuda.device_count() > 1:
-        model = create_parallel_model()
-    else:
-        model = create_model()
-    model.to(device)
-    optimizer = optim.Adam(model.parameters())
-    optimizer.step()
-
-    with mlflow.start_run():
-        # Create a SummaryWriter to write TensorBoard events locally
-        events_output_dir = tempfile.mkdtemp()
-        writer = SummaryWriter(events_output_dir)
-        click.echo(click.style(f'Writing TensorBoard events locally to {events_output_dir}\n', fg='blue'))
-
-        # Start training
-        runtime = time.time()
-        for epoch in range(1, epochs + 1):
-            train(use_cuda, model, epoch, optimizer, log_interval, train_loader, writer)
-            test(use_cuda, model, epoch, test_loader, writer)
-        device = 'GPU' if use_cuda else 'CPU'
-        click.echo(click.style(f'{device} Run Time: {str(time.time() - runtime)} seconds', fg='green'))
-
-        # Closing writer to allow for the model to be logged
-        writer.close()
-
-        # Log the model to mlflow
-        click.echo(click.style('Logging model to mlflow...', fg='blue'))
-        mlflow.pytorch.log_model(model, 'models')
-
-        # Log hardware and software
-        log_sys_intel_conda_env()
-
-        # Upload the TensorBoard event logs as a run artifact
-        click.echo(click.style('Uploading TensorBoard events as a run artifact...', fg='blue'))
-        mlflow.log_artifacts(events_output_dir, artifact_path='events')
-        click.echo(click.style(f'\nLaunch TensorBoard with:\ntensorboard --logdir={os.path.join(mlflow.get_artifact_uri(), "events")}', fg='blue'))
-
-
-def set_pytorch_random_seeds(seed, use_cuda):
-    torch.manual_seed(seed)
-    if use_cuda:
-        torch.cuda.manual_seed(seed)
-        torch.cuda.manual_seed_all(seed)  # For multiGPU
-        torch.backends.cudnn.deterministic = True
-        torch.backends.cudnn.benchmark = False
-
-
-if __name__ == '__main__':
-    traceback.install()
-    click.echo(click.style(f'Num GPUs Available: {torch.cuda.device_count()}', fg='blue'))
-
-    start_training()
+        print(f'[bold blue]Using {torch.cuda.device_count()} GPUs!')
+
+    parser = pl.Trainer.add_argparse_args(parent_parser=parser)
+    parser = LightningMNISTClassifier.add_model_specific_args(parent_parser=parser)
+
+    mlflow.pytorch.autolog()
+    # log conda env and system information
+    log_sys_intel_conda_env()
+    # parse cli arguments
+    args = parser.parse_args()
+    dict_args = vars(args)
+
+    set_general_random_seeds(dict_args['general_seed'])
+    set_pytorch_random_seeds(dict_args['pytorch_seed'], True)
+
+    if "accelerator" in dict_args:
```

Comment:
Can't this be done in easier ways?

Reply:
You meant via `seed_everything`?

Reply:
No. Checking the dict for the argument + replacing none etc

Reply:
Ah well, I currently have no idea how to shorten this

Reply:
Just looks really ugly. I'll take a look at it again when I find the time.

Reply:
Ah `my_dict.get(mykey)` does return `None` if key does not exist